File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/ObservationDefinition.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class ObservationDefinition : DomainResource, IResource
{
    public ObservationDefinition_QuantitativeDetails? QuantitativeDetails { get; set; }
    public Base.CodeableConcept[]? Category { get; set; }
    public Base.CodeableConcept? Method { get; set; }
    public Base.ResourceReference? ValidCodedValueSet { get; set; }
    public ObservationDefinition_QualifiedInterval[]? QualifiedInterval { get; set; }
    public Base.ResourceReference? AbnormalCodedValueSet { get; set; }
    public required Base.CodeableConcept Code { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public string[]? PermittedDataType { get; set; }
    public bool? MultipleResultsAllowed { get; set; }
    public Base.ResourceReference? NormalCodedValueSet { get; set; }
    public string? PreferredReportName { get; set; }
    public Base.ResourceReference? CriticalCodedValueSet { get; set; }

    public class ObservationDefinition_QuantitativeDetails : BackboneElement
    {
        public Base.CodeableConcept? Unit { get; set; }
        public Base.CodeableConcept? CustomaryUnit { get; set; }
        public string? ConversionFactor { get; set; }
        public int? DecimalPrecision { get; set; }
    }

    public class ObservationDefinition_QualifiedInterval : BackboneElement
    {
        public Base.Range? Age { get; set; }
        public Base.Range? Range { get; set; }
        public string? Gender { get; set; }
        public Base.CodeableConcept? Context { get; set; }
        public string? Category { get; set; }
        public Base.CodeableConcept[]? AppliesTo { get; set; }
        public string? Condition { get; set; }
        public Base.Range? GestationalAge { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/ObservationDefinition.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import CodeableConcept
from base import Range
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class ObservationDefinitionQuantitativeDetails(BackboneElement):
    unit: Optional[CodeableConcept] = None
    customary_unit: Optional[CodeableConcept] = None
    conversion_factor: Optional[float] = None
    decimal_precision: Optional[int] = None

@dataclass(kw_only=True)
class ObservationDefinitionQualifiedInterval(BackboneElement):
    age: Optional[Range] = None
    range: Optional[Range] = None
    gender: Optional[str] = None
    context: Optional[CodeableConcept] = None
    category: Optional[str] = None
    applies_to: Optional[List[CodeableConcept]] = field(default_factory=list)
    condition: Optional[str] = None
    gestational_age: Optional[Range] = None

@dataclass(kw_only=True)
class ObservationDefinition(DomainResource):
    quantitative_details: Optional[ObservationDefinitionQuantitativeDetails] = None
    category: Optional[List[CodeableConcept]] = field(default_factory=list)
    method: Optional[CodeableConcept] = None
    valid_coded_value_set: Optional[Reference] = None
    qualified_interval: Optional[List[ObservationDefinitionQualifiedInterval]] = field(default_factory=list)
    abnormal_coded_value_set: Optional[Reference] = None
    code: CodeableConcept
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    permitted_data_type: Optional[List[str]] = field(default_factory=list)
    multiple_results_allowed: Optional[bool] = None
    normal_coded_value_set: Optional[Reference] = None
    preferred_report_name: Optional[str] = None
    critical_coded_value_set: Optional[Reference] = None